projects/plugins/debug-helper/modules/class-idc-simulator.php

```diff
@@ -0,0 +1,238 @@
+<?php // phpcs:disable WordPress.PHP.DevelopmentFunctions.error_log_print_r
+/**
+ * Plugin Name: IDC Simulator
+ * Description: Cause an IDC on your site without having to clone it.
+ * Author: Bestpack
+ * Version: 1.0
+ * Text Domain: jetpack
+ *
+ * @package automattic/jetpack-debug-helper
+ */
+
+/**
+ * Class Broken_Token
+ */
+class IDC_Simulator {
+	/**
+	 * Notice type.
+	 *
+	 * @var mixed|string
+	 */
+	public $notice_type = '';
+
+	/**
+	 * Whether to enable siteurl spoofing.
+	 *
+	 * @var Boolean
+	 */
+	protected $idc_simulation_enabled = false;
+
+	/**
+	 * What URL to use instead of the real siteurl.
+	 *
+	 * @var Boolean
+	 */
+	protected $idc_siteurl;
+
+	/**
+	 * Options.
+	 */
+	const STORED_OPTIONS_KEY = 'idc_simulator_stored_options';
+
+	/**
+	 * IDC_Simulator constructor.
+	 */
+	public function __construct() {
+		add_action( 'admin_menu', array( $this, 'idc_simulator_register_submenu_page' ), 1000 );
+
+		add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_scripts' ) );
+
+		// Stored options.
+		add_action( 'admin_post_store_current_options', array( $this, 'admin_post_store_current_options' ) );
+
+		$settings                     = $this->get_stored_settings();
+		$this->idc_simulation_enabled = 1 === intval( $settings['idc_simulation'] ) ?
```

Comment:
If the option is empty, `get_stored_settings()` will return false, and `$settings['idc_simulation']` will generate a PHP notice:

> PHP Notice:  Trying to access array offset on value of type bool in .../plugins/debug-helper/modules/class-idc-simulator.php on line 54

Should we check whether `$settings['idc_simulation']` exists before trying to use it?

Reply:
I have added `wp_parse_args` to add sane defaults, thanks for the catch!